### ISSUE TITLE
Minified file not truncated

### ISSUE THREAD

**Issue by wagnerand** [mozilla/addons-code-manager#1336]
### Describe the problem and steps to reproduce it:

https://code.addons.mozilla.org/en-US/browse/982178/versions/4923182/?path=contentscript.js is very slow to load, even triggers a "slow js" warning in the browser, because it's minified.

### What happened?

The file was not truncated.

### What did you expect to happen?

The file would show as truncated.

**Comment by bobsilverberg** [comment]
This was fixed via https://github.com/mozilla/addons-code-manager/issues/1328, but that won't land on prod until the push of 2020-01-09, which is why you are not seeing it on prod. You could try checking on dev to see if it's working as expected there.

**Comment by bobsilverberg** [comment]
Actually, sorry, this is about truncation, not syntax highlighting. If the file is not above the specified line count, then it won't be truncated, no matter how wide it is.

We haven't discussed truncating wide files, only disabling highlighting for them. Is this something you think we should do? Perhaps we should wait to see if disabling highlighting is enough for now?

**Comment by wagnerand** [comment]
We should truncate minified files. Reviewers usually can't review them anyway.

**Comment by bobsilverberg** [comment]
> We should truncate minified files. Reviewers usually can't review them anyway.

These files which are not being truncated do not have a large number of lines, but rather have very long lines. So in this case it would make sense to truncate the *lines* themselves, and not just the length of the file. Do you agree?

**Comment by wagnerand** [comment]
Agreed.

**Comment by willdurand** [comment]
> These files which are not being truncated do not have a large number of lines, but rather have very long lines. So in this case it would make sense to truncate the _lines_ themselves, and not just the length of the file. Do you agree?

We should think in terms of number of characters instead then (I think)

**Comment by bobsilverberg** [comment]
> > These files which are not being truncated do not have a large number of lines, but rather have very long lines. So in this case it would make sense to truncate the _lines_ themselves, and not just the length of the file. Do you agree?
> 
> We should think in terms of number of characters instead then (I think)

@willdurand and I discussed this, and I'm not sure that number of characters will work. There is a width that causes problems, regardless of how long the file is, and, coversely, a file with a lot of short lines is also not a problem, so conflating the two checks might not work.

Another issue is the fact that if a file has too many lines, we truncate it by only displaying the first `x` lines, but if a file has lines that are too wide, we must truncate those lines themselves, which is a separate operation.

**Comment by bobsilverberg** [comment]
We have actually decided to try going the route of truncating based on total number of characters in a file, which is https://github.com/mozilla/addons-code-manager/issues/1347, so that is now a duplicate of this. I will leave this in the 2020.01.09 milestone so it can be verified that landing #1347 does in fact fix this.

**Comment by ioanarusiczki** [comment]
@bobsilverberg Is https://github.com/mozilla/addons-code-manager/issues/1347 going to land this milestone ?

**Comment by ioanarusiczki** [comment]
@wagnerand For this file on dev https://code.addons-dev.allizom.org/en-US/browse/502868/versions/1688187/?path=src/js/Grammarly-signin.js I could scroll up and down without having a warning - the highlighting is disabled now, so perhaps that helps too ( I hope my example is similar with the one in production). 
The truncation can be verified with the next coming milestone. So, I'm not going to mark this verified yet.

**Comment by bobsilverberg** [comment]
Thanks @ioanarusiczki. Your observations are correct. The fact that the file isn't being highlighted should partially address the problem, but the file will also be truncated after #1347 lands, which won't happen until next week.

**Comment by ioanarusiczki** [comment]
I verified the file again on dev with FF72(Win10) - https://code.addons-dev.allizom.org/en-US/browse/502868/versions/1688187/?allowSlowPages=false&path=src%2Fjs%2FGrammarly-signin.js on dev with FF72(Win10). 
- the file truncated loads without delays. 
- the highlighting was enabled for truncated files so right now to view the original file will take some time to load. 

![original file - takes time](https://user-images.githubusercontent.com/33448286/72266945-ec0b2800-3627-11ea-9267-708e592330a1.gif)


@bobsilverberg If reviewers don't review such files the scope is to make navigation through the file tree more efficient which right now it's happening. I'm going to mark this verified.
